core/src/main/java/org/apache/calcite/plan/RelOptCluster.java

```diff
@@ -115,6 +116,10 @@ public RelOptPlanner getPlanner() {
     return planner;
   }
 
+  public void setPlanner(RelOptPlanner planner) {
```

Comment:
This is a HACK to reset planner for all nodes in a logical plan. Alternative may be to use a RelShuttle to make a new plan with the desired planner. Not sure if we have a better way.

Reply:
Can you explain why we have to reset planner? Thanks!

Reply:
After initial mapping of Pig logical plan to Calcite logical plan, I need to write an optimizer rule to optimize the plan. So need to write a new planner for to optimize the plan and assign this new planner to all rel nodes in the plan.